File: 739-daily-temperatures/daily-temperatures.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function (temperatures) {
    const response = Array(temperatures.length).fill(0);
    const stack = [];
    for (const [index, value] of temperatures.entries()) {
        while (stack.length > 0 && stack[stack.length - 1][0] < value) {
            const curr = stack.pop();
            response[curr[1]] = index - curr[1];
        }

        stack.push([value, index]);
    }

    return response;
};

File: 739-daily-temperatures/daily-temperatures.cs
public class Solution {
    public int[] DailyTemperatures(int[] temperatures) {
        Stack<int> stack = new Stack<int>();
        int[] result = new int[temperatures.Length];
        result[^1] = 0;

        for (int i = 0; i < temperatures.Length; i++){
            while(stack.Count> 0 && temperatures[stack.Peek()] < temperatures[i]){
                int index = stack.Pop();
                result[index] = i - index;
            }
            stack.Push(i);
        }

        return result;
    }
}